packages/material-ui-lab/src/SliderStyled/SliderStyled.test.js

```diff
@@ -522,14 +523,55 @@ describe('<Slider />', () => {
   });
 
   describe('prop: valueLabelDisplay', () => {
-    it('should always display the value label', () => {
-      const { getByRole, setProps } = render(<Slider valueLabelDisplay="auto" value={50} />);
+    it('should always display the value label according to on and off', () => {
+      const { getByRole, setProps } = render(<Slider valueLabelDisplay="on" value={50} />);
       const thumb = getByRole('slider');
-      expect(thumb.textContent).to.equal('50');
+      expect(thumb.firstChild).to.have.class('MuiSlider-valueLabelOpen');
+
       setProps({
         valueLabelDisplay: 'off',
       });
-      expect(thumb.textContent).to.equal('');
+
+      const newThumb = getByRole('slider');
+      expect(newThumb.firstChild).to.equal(null);
+    });
+
+    it('should display the value label only on hover for auto', () => {
+      const { getByRole } = render(<Slider valueLabelDisplay="auto" value={50} />);
+      const thumb = getByRole('slider');
+      expect(thumb.firstChild).not.to.have.class('MuiSlider-valueLabelOpen');
+
+      fireEvent.mouseOver(thumb);
+
+      expect(thumb.firstChild).to.have.class('MuiSlider-valueLabelOpen');
+    });
+
+    it('should be respected when using custom value label', () => {
+      function ValueLabelComponent(props) {
+        const { value, open } = props;
+        return (
+          <span data-testid="value-label" className={clsx({ open })}>
+            {value}
+          </span>
+        );
+      }
+      ValueLabelComponent.propTypes = { value: PropTypes.number };
+
+      const screenResult = render(
```

Comment:
`screen` is reserved for the actual screen. You can just destructure what you need as far as I can tell.